dagger/utilities/dbt_config_parser.py

```diff
@@ -0,0 +1,208 @@
+from os import path
+from os.path import join
+from typing import Union, Tuple, List, Dict
+
+import json
+import yaml
+
+ATHENA_IO_BASE = {"type": "athena"}
+S3_IO_BASE = {"type": "s3"}
+
+
+
+class DBTConfigParser:
+    """
+    Module that parses the manifest.json file generated by dbt and generates the dagger inputs and outputs for the respective dbt model
+    """
+
+    def __init__(self, default_config_parameters: dict):
+        self._default_data_bucket = default_config_parameters["data_bucket"]
+        self._dbt_project_dir = default_config_parameters.get("project_dir", None)
+        dbt_manifest_path = path.join(self._dbt_project_dir, "target", "manifest.json")
+        self._dbt_profile_dir = default_config_parameters.get("profile_dir", None)
+        dbt_profile_path = path.join(self._dbt_profile_dir, "profiles.yml")
+
+        with open(dbt_manifest_path, "r") as f:
+            data = f.read()
+        self._manifest_data = json.loads(data)
+        profile_yaml = yaml.safe_load(open(dbt_profile_path, "r"))
+        prod_dbt_profile = profile_yaml[self._dbt_project_dir.split("/")[-1]]["outputs"]["data"]
+        self._default_data_dir = prod_dbt_profile.get(
+            "s3_data_dir"
+        ) or prod_dbt_profile.get("s3_staging_dir")
+
+    def generate_io(self, model_name: str) -> Tuple[List[Dict], List[Dict]]:
+        """
+        Generates the dagger inputs and outputs for the respective dbt model
+        Args:
+            model_name: name of the dbt model
+
+        Returns:
+            tuple[List[Dict], List[Dict]]: dagger inputs and outputs for the respective dbt model
+
+        """
+        model_parents = self._get_dbt_model_parents(model_name)
+        model_dagger_inputs = self.generate_dagger_inputs(model_parents)
+        model_dagger_outputs = self.generate_dagger_outputs(
+            model_parents["model_name"],
+            model_parents["schema"],
+            model_parents["relative_s3_path"],
+        )
+
+        return model_dagger_inputs, model_dagger_outputs
+
+    def parse_dbt_staging_model(self, dbt_staging_model: str) -> Tuple[str, str]:
+        """
+        Parses the dbt staging model to get the core schema and table name
+        Args:
+            dbt_staging_model: name of the DBT staging model
+
+        Returns:
+            Tuple[str, str]: core schema and table name
+
+        >>> parse_dbt_staging_model("schema_name__table")
+        ('schema_name', 'table')
+        >>> parse_dbt_staging_model("another_schema__another_table")
+        ('another_schema', 'another_table')
+        """
+        _model_split, core_table = dbt_staging_model.split("__")
+        core_schema = _model_split.split("_")[-1]
+
+        return core_schema, core_table
```

Comment:
This also seems to be a very Choco customized solution and heavily depending on how we name our staging models. Is there any way to do this better? I don't have a suggestion, just asking.